src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,118 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
+    const queryKey = [
+        `validationRules`,
+        {
+            params: {
+                dataSet: datasetId,
+                fields: [
+                    'id',
+                    'importance',
+                    'operator',
+                    'leftSide',
+                    'rightSide',
+                    'displayInstruction',
+                    'displayDescription',
+                    'displayName',
+                ],
+            },
+        },
+    ]
+
+    return queryKey
+}
+
+const getValidationQueryKey = (
+    datasetId,
+    { periodId, orgUnitId, categoryComboId, categoryOptionIds }
+) => {
+    const params = {
+        pe: periodId,
+        ou: orgUnitId,
+        cc: categoryComboId,
+        cp: categoryOptionIds?.join(';'),
+    }
+
+    const validationQueryKey = [
+        `validation/dataSet/${datasetId}`,
+        {
+            params,
+        },
+    ]
+    return validationQueryKey
+}
+export const useValidationResult = () => {
+    const [datasetId] = useDataSetId()
+
+    const [periodId] = usePeriodId()
+    const [orgUnitId] = useOrgUnitId()
+    const {
+        attributeCombo: categoryComboId,
+        attributeOptions: categoryOptionIds,
+    } = useApiAttributeParams()
+
+    const validationQueryKey = getValidationQueryKey(datasetId, {
+        periodId,
+        orgUnitId,
+        categoryComboId,
+        categoryOptionIds,
+    })
+    const validationMetaDataQuery = getValidationMetaDataQueryKey(datasetId)
+
+    const activeMutations = useIsMutating({ mutationKey: validationQueryKey })
+    const enabled = activeMutations === 0
```

Comment:
Maybe `useIsFetching` would be more semantic here?

Reply:
Yeah I suppose that's correct -- is your intention to disable these queries while a new data value is being sent to the server, or something else? (If it's the first, maybe you could use `isMutating` with some [mutation filters](https://tanstack.com/query/v4/docs/guides/filters#mutation-filters) to pause these queries while syncing data)

Reply:
yeah I think that's the intention. I added after a remark from @Mohammer5 

> We probably should check that there are no active mutations (see the useDataValueSet hook).

So I copied the logic from useDataValueSet. But I think the mutationKey I am checking against doesn't make sense.

Reply:
Yes, so using the `validationQueryKey` doesn't make much sense here, it'd have to be the same query key that the `useDataValueSet` hook uses. I suppose you could simply use the `useDataValueSet`'s `isLoading` state here, it shouldn't trigger another fetch.

Maybe it makes more sense to disable the "Run validation" button instead though.

Reply:
I  used `useIsMutating` with the key from `useDataValueSetQueryKey` to get the behaviour we're after. Otherwise if I use `useDataValueSet`, it disables the button when refetching the data element after it was updated (not while it's updating). Please have at [the change](https://github.com/dhis2/data-entry-app/pull/93/commits/f855db6f87e814625503ec55bfeba35ca77a6bc6) and let me know if this looks right.

Reply:
Hm, I don't think disabling based on the `dataValueSets` fetch is correct -- I think we should disable it based on the `dataValues` mutations, which `use-data-value-set.js` seems to be lacking too

Based on the mutation filters docs linked above, I think this might look like `useIsMutating({ mutationKey: ['dataValues'] })`

Reply:
Oh, that's... confusing 🤔

Reply:
I created a subtask to look more into this: https://dhis2.atlassian.net/browse/TECH-1289 .. so just this PR can be unblocked and I can raise PRs for the next bits related to servr-side validation

Reply:
Great, thanks 🙂👍